librz/core/cmd/cmd_cmp.c

```diff
@@ -473,384 +103,324 @@ static void __core_cmp_bits(RzCore *core, ut64 addr) {
-		switch (sz) {
-		case '1': { // "cv1"
-			ut8 n = (ut8)rz_num_math(core->num, input + 2);
-			core->num->value = 1;
-			if (block[0] == n) {
-				rz_cons_printf("0x%08" PFMT64x "\n", core->offset);
-				core->num->value = 0;
-			}
-			break;
-		}
-		case '2': { // "cv2"
-			ut16 n = (ut16)rz_num_math(core->num, input + 2);
-			core->num->value = 1;
-			if (core->blocksize >= 2 && *(ut16 *)block == n) {
-				rz_cons_printf("0x%08" PFMT64x "\n", core->offset);
-				core->num->value = 0;
-			}
-			break;
-		}
-		case '4': { // "cv4"
-			ut32 n = (ut32)rz_num_math(core->num, input + 2);
-			core->num->value = 1;
-			if (core->blocksize >= 4 && *(ut32 *)block == n) {
-				rz_cons_printf("0x%08" PFMT64x "\n", core->offset);
-				core->num->value = 0;
-			}
-			break;
-		}
-		case '8': { // "cv8"
-			ut64 n = (ut64)rz_num_math(core->num, input + 2);
-			core->num->value = 1;
-			if (core->blocksize >= 8 && *(ut64 *)block == n) {
-				rz_cons_printf("0x%08" PFMT64x "\n", core->offset);
-				core->num->value = 0;
-			}
-			break;
-		}
-		default:
-		case '?':
-			eprintf("Usage: cv[1248] [num]\n"
-				"Show offset if current value equals to the one specified\n"
-				" /v 18312   # serch for a known value\n"
-				" dc\n"
-				" cv4 18312 @@ hit*\n"
-				" dc\n");
-			break;
-		}
-	} break;
-	case 'V': { // "cV"
-		int sz = input[1];
-		if (sz == ' ') {
-			switch (rz_config_get_i(core->config, "asm.bits")) {
-			case 8: sz = '1'; break;
-			case 16: sz = '2'; break;
-			case 32: sz = '4'; break;
-			case 64: sz = '8'; break;
-			default: sz = '4'; break; // default
-			}
-		} else if (sz == '?') {
-			eprintf("Usage: cV[1248] [addr] @ addr2\n"
-				"Compare n bytes from one address to current one and return in $? 0 or 1\n");
-		}
-		sz -= '0';
-		if (sz > 0) {
-			ut64 at = rz_num_math(core->num, input + 2);
-			ut8 buf[8] = { 0 };
-			rz_io_read_at(core->io, at, buf, sizeof(buf));
-			core->num->value = memcmp(buf, core->block, sz) ? 1 : 0;
-		}
-		break;
-	}
-	case 'l': // "cl"
-		if (strchr(input, 'f')) {
-			rz_cons_flush();
-		} else if (input[1] == 0) {
-			rz_cons_fill_line();
-			// rz_cons_clear_line (0);
-		} else if (!strchr(input, '0')) {
-			rz_cons_clear00();
+// cb
+RZ_IPI RzCmdStatus rz_cmd_cmp_bytes_handler(RzCore *core, int argc, const char **argv, RzCmdStateOutput *state) {
+	RzCmdStatus ret = RZ_CMD_STATUS_ERROR;
+	ut64 sz = rz_num_math(core->num, argv[2]);
+	if (sz > 8) {
+		RZ_LOG_ERROR("Cannot compare more than 8 bytes. Use the c command instead.\n");
+		return ret;
+	}
+
+	ut64 num = rz_num_math(core->num, argv[1]);
+	ut64 mask = -1;
+	mask >>= 8 - sz;
+	ut64 valid_num = num & mask;
+	RzCompareData *cmp = rz_core_cmp_mem_data(core, core->offset, (ut8 *)&valid_num, sz);
+	if (!cmp) {
+		goto end;
+	}
+	if (cmp->same) {
+		core->num->value = 0;
+	} else {
+		core->num->value = 1;
+	}
+
+	int val = rz_core_cmp_print(core, cmp, state);
+	if (val != -1) {
+		ret = RZ_CMD_STATUS_OK;
+	}
+
+end:
+	rz_core_cmp_free(cmp);
+	return ret;
```

Comment:
I suggest to create a small function that accepts `cmp` and sets core->num->value based on `cmp`, prints it and, free it and then returns the right RZ_CMD_STATUS_ value. You can use this in most handlers to avoid duplicate code and ensure more consistency (e.g. in some handlers you are setting core->num->value, in others you aren't).

Reply:
As per the discussion on Mattermost, I am removing setting the `core->num->value` where it is not necessary.